src/joystick/hidapi/SDL_hidapi_switch.c

```diff
@@ -943,23 +943,23 @@ static Uint8 RemapButton(SDL_DriverSwitch_Context *ctx, Uint8 button)
         /* Use button positions */
         if (ctx->m_bIsGameCube) {
             switch (button) {
-            case SDL_GAMEPAD_BUTTON_B:
-                return SDL_GAMEPAD_BUTTON_X;
-            case SDL_GAMEPAD_BUTTON_X:
-                return SDL_GAMEPAD_BUTTON_B;
+            case SDL_GAMEPAD_BUTTON_EAST:
+                return SDL_GAMEPAD_BUTTON_WEST;
+            case SDL_GAMEPAD_BUTTON_WEST:
+                return SDL_GAMEPAD_BUTTON_EAST;
             default:
                 break;
             }
         } else {
             switch (button) {
-            case SDL_GAMEPAD_BUTTON_A:
-                return SDL_GAMEPAD_BUTTON_B;
-            case SDL_GAMEPAD_BUTTON_B:
-                return SDL_GAMEPAD_BUTTON_A;
-            case SDL_GAMEPAD_BUTTON_X:
-                return SDL_GAMEPAD_BUTTON_Y;
-            case SDL_GAMEPAD_BUTTON_Y:
-                return SDL_GAMEPAD_BUTTON_X;
+            case SDL_GAMEPAD_BUTTON_SOUTH:
+                return SDL_GAMEPAD_BUTTON_EAST;
+            case SDL_GAMEPAD_BUTTON_EAST:
+                return SDL_GAMEPAD_BUTTON_SOUTH;
+            case SDL_GAMEPAD_BUTTON_WEST:
+                return SDL_GAMEPAD_BUTTON_NORTH;
+            case SDL_GAMEPAD_BUTTON_NORTH:
+                return SDL_GAMEPAD_BUTTON_WEST;
             default:
```

Comment:
Shouldn't this remapping be removed?

Reply:
`m_bUseButtonLabels` has been removed on other hidapi devices, but not here.

Reply:
Yep, we still need to handle GameCube form factors and the classic Nintendo Online controllers.